src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyNonCacheableResponse.java

```diff
@@ -0,0 +1,262 @@
+import org.reaktivity.nukleus.http_cache.internal.types.stream.AbortFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.BeginFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.DataFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.EndFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.HttpBeginExFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.ResetFW;
+import org.reaktivity.nukleus.http_cache.internal.types.stream.WindowFW;
+
+import static java.lang.System.currentTimeMillis;
+import static org.reaktivity.nukleus.http_cache.internal.HttpCacheConfiguration.DEBUG;
+import static org.reaktivity.nukleus.http_cache.internal.stream.util.HttpHeadersUtil.getHeader;
+
+final class HttpCacheProxyNonCacheableResponse extends HttpCacheProxyResponse
+{
+    private final HttpCacheProxyFactory streamFactory;
+
+    private MessageConsumer streamState;
+
+    private HttpCacheProxyRequest request;
+    private final MessageConsumer connectReplyThrottle;
+    private final long connectRouteId;
+    private final long connectReplyStreamId;
+
+    private long acceptInitialId;
+    private int acceptReplyBudget;
+
+    private long groupId;
+    private int padding;
+
+    private long traceId;
+
+    HttpCacheProxyNonCacheableResponse(
+        HttpCacheProxyFactory httpCacheProxyFactory,
+        HttpCacheProxyRequest request,
+        MessageConsumer connectReplyThrottle,
+        long connectRouteId,
+        long connectReplyId,
+        long acceptInitialId)
+    {
+        this.streamFactory = httpCacheProxyFactory;
+        this.request = request;
+        this.connectReplyThrottle = connectReplyThrottle;
+        this.connectRouteId = connectRouteId;
+        this.connectReplyStreamId = connectReplyId;
+        this.acceptInitialId = acceptInitialId;
+        this.streamState = this::onResponseMessage;
+    }
+
+    public void onResponseMessage(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        switch (msgTypeId)
+        {
+            case BeginFW.TYPE_ID:
+                final BeginFW begin = this.streamFactory.beginRO.wrap(buffer, index, index + length);
+                handleBegin(begin);
+                break;
+            case WindowFW.TYPE_ID:
+                final WindowFW window = streamFactory.windowRO.wrap(buffer, index, index + length);
+                onWindowWhenProxying(window);
+                break;
+            case ResetFW.TYPE_ID:
+                final ResetFW reset = streamFactory.resetRO.wrap(buffer, index, index + length);
+                onResetWhenProxying(reset);
+                break;
+            default:
+                // ignore
+                break;
+        }
+    }
+
+    void handleStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        streamState.accept(msgTypeId, buffer, index, length);
+    }
```

Comment:
Remove this method, use `onResponseMessage` instead.

Reply:
Done.